pkg/operator/operator_deployment_relication.go

```diff
@@ -0,0 +1,215 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Ewout Prangsma
+//
+
+package operator
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	kwatch "k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+
+	api "github.com/arangodb/kube-arangodb/pkg/apis/replication/v1alpha"
+	"github.com/arangodb/kube-arangodb/pkg/metrics"
+	"github.com/arangodb/kube-arangodb/pkg/replication"
+	"github.com/arangodb/kube-arangodb/pkg/util/k8sutil"
+)
+
+var (
+	deploymentReplicationsCreated  = metrics.MustRegisterCounter("controller", "deployment_replications_created", "Number of deployment replications that have been created")
+	deploymentReplicationsDeleted  = metrics.MustRegisterCounter("controller", "deployment_replications_deleted", "Number of deployment replications that have been deleted")
+	deploymentReplicationsFailed   = metrics.MustRegisterCounter("controller", "deployment_replications_failed", "Number of deployment replications that have failed")
+	deploymentReplicationsModified = metrics.MustRegisterCounter("controller", "deployment_replications_modified", "Number of deployment replication modifications")
+	deploymentReplicationsCurrent  = metrics.MustRegisterGauge("controller", "deployment_replications", "Number of deployment replications currently being managed")
+)
+
+// run the deployment replications part of the operator.
+// This registers a listener and waits until the process stops.
+func (o *Operator) runDeploymentReplications(stop <-chan struct{}) {
+	rw := k8sutil.NewResourceWatcher(
+		o.log,
+		o.Dependencies.CRCli.ReplicationV1alpha().RESTClient(),
+		api.ArangoDeploymentReplicationResourcePlural,
+		o.Config.Namespace,
+		&api.ArangoDeploymentReplication{},
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    o.onAddArangoDeploymentReplication,
+			UpdateFunc: o.onUpdateArangoDeploymentReplication,
+			DeleteFunc: o.onDeleteArangoDeploymentReplication,
+		})
+
+	o.Dependencies.DeploymentReplicationProbe.SetReady()
+	rw.Run(stop)
+}
+
+// onAddArangoDeploymentReplication deployment replication addition callback
+func (o *Operator) onAddArangoDeploymentReplication(obj interface{}) {
+	o.Dependencies.LivenessProbe.Lock()
+	defer o.Dependencies.LivenessProbe.Unlock()
+
+	apiObject := obj.(*api.ArangoDeploymentReplication)
+	o.log.Debug().
+		Str("name", apiObject.GetObjectMeta().GetName()).
+		Msg("ArangoDeploymentReplication added")
+	o.syncArangoDeploymentReplication(apiObject)
+}
+
+// onUpdateArangoDeploymentReplication deployment replication update callback
+func (o *Operator) onUpdateArangoDeploymentReplication(oldObj, newObj interface{}) {
+	o.Dependencies.LivenessProbe.Lock()
```

Comment:
How is oldObj used?  I only see it in the function declaration (or I am blind).

Reply:
It is not used. The reason it is still there, because this function is a callback with a predefined signature.